gpkit/tools/tools.py

```diff
@@ -29,6 +29,95 @@ def te_exp_minus1(posy, nterm):
         res += posy**i / factorial_denom
     return res
 
+def te_secant(var, nterm):
+    """Taylor expansion of secant(var).
+
+    Arguments
+    ---------
+    var : gpkit.monomial
+      Variable or expression argument
+    nterm : int
+        Number of terms in resulting Taylor expansion
+
+    Returns
+    -------
+    gpkit.Posynomial
+        Taylor expansion of secant(x), carried to nterm terms
+    """
+    if nterm < 1:
+        raise ValueError("Unexpected number of terms, nterm=%s" % nterm)
+
+    # The first 12 Euler Numbers
+    E2n = np.asarray([1,
+                      5,
+                      61,
+                      1385,
+                      50521,
+                      2702765,
+                      199360981,
+                      19391512145,
+                      2404879675441,
+                      370371188237525,
+                      69348874393137901,
+                      15514534163557086905])
+    if nterm > 12:
+        n_extend = np.asarray(range(13, nterm+1))
+        E2n_add = (8 * np.sqrt(n_extend/np.pi)
+                   * (4*n_extend/(np.pi * np.exp(1)))**(2*n_extend))
+        E2n = np.append(E2n, E2n_add)
+
+    res = 1
+    factorial_denom = 1
+    for i in range(1, nterm + 1):
+        factorial_denom *= ((2*i)*(2*i-1))
+        res += E2n[i-1] / factorial_denom * var**(2*i)
```

Comment:
Lol. Well, probably best to just put period after all those numbers to be safe. Adding a float to the denominator calculation on line 72 would also be good. It would be good to remove some of the numbers from the table and make sure your unit test catches the integer division error.